application/client.js

```diff
@@ -152,7 +152,14 @@ module.exports = HttpGateway.extend({
                     try {
                         responseData = JSON.parse(responseText);
                     } catch (e) {
-                        responseData = responseText;
```

Comment:
I'm not sure this belongs here. A couple of questions:

- WHY is the status code zero?
- Is the OPTIONS request part of the browser? If not, what causes it?